metrics/internal/collectors/ceph-blocklist.go

```diff
@@ -6,28 +6,48 @@ import (
 
 	"github.com/prometheus/client_golang/prometheus"
 	internalcache "github.com/red-hat-storage/ocs-operator/metrics/internal/cache"
+	"github.com/red-hat-storage/ocs-operator/metrics/internal/options"
+	"golang.org/x/net/context"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
 	"k8s.io/klog"
 )
 
 var _ prometheus.Collector = &CephBlocklistCollector{}
 
 type CephBlocklistCollector struct {
+	kubeClient            clientset.Interface
 	CephBlocklistStore    *internalcache.CephBlocklistStore
 	PersistentVolumeStore *internalcache.PersistentVolumeStore
 	// Metrics Descriptors
 	NodeRBDBlocklist *prometheus.Desc
 }
 
-func NewCephBlocklistCollector(blocklistStore *internalcache.CephBlocklistStore, pvStore *internalcache.PersistentVolumeStore) *CephBlocklistCollector {
+func NewCephBlocklistCollector(opts *options.Options, blocklistStore *internalcache.CephBlocklistStore, pvStore *internalcache.PersistentVolumeStore) *CephBlocklistCollector {
 	return &CephBlocklistCollector{
+		kubeClient:            clientset.NewForConfigOrDie(opts.Kubeconfig),
 		CephBlocklistStore:    blocklistStore,
 		PersistentVolumeStore: pvStore,
 		NodeRBDBlocklist: prometheus.NewDesc(
 			prometheus.BuildFQName(namespace, "", "rbd_client_blocklisted"),
-			"State of the rbd client on a node, 0 = Unblocked, 1 = Blocked", []string{"node_name"}, nil),
+			"State of the rbd client on a node, 0 = Unblocked, 1 = Blocked", []string{"node"}, nil),
 	}
 }
 
+func (c *CephBlocklistCollector) getNodes() (map[string]bool, error) {
```

Comment:
`getBlockedNodes` might be a better name for what this function does.

Reply:
this only gets all the node on the system, and initializes all the values as false